uxarray/core/dataset.py

```diff
@@ -327,6 +327,189 @@ def from_healpix(
 
         return cls.from_xarray(ds, uxgrid, {face_dim: "n_face"})
 
+    @classmethod
+    def _wrap_groupby_result(
+        cls, result, original_uxgrid, original_source_datasets=None
+    ):
+        """Factory method to wrap groupby results with uxgrid preservation."""
+        if isinstance(result, xr.Dataset):
+            return cls(
+                result, uxgrid=original_uxgrid, source_datasets=original_source_datasets
+            )
+        elif isinstance(result, xr.DataArray):
+            from uxarray.core.dataarray import UxDataArray
+
+            return UxDataArray(result, uxgrid=original_uxgrid)
+        return result
+
+    def groupby(self, group, squeeze: bool = False, restore_coord_dims: bool = None):
+        """Get the standard xarray groupby object"""
+        groupby_obj = super().groupby(
+            group, squeeze=squeeze, restore_coord_dims=restore_coord_dims
+        )
+
+        # Store references to preserve uxgrid info
+        original_uxgrid = self.uxgrid
+        original_source_datasets = self.source_datasets
+
+        # Modify the aggregation methods to preserve uxgrid
+        for method_name in [
+            "mean",
+            "sum",
+            "min",
+            "max",
+            "std",
+            "var",
+            "median",
+            "count",
+        ]:
```

Comment:
I wonder if there is any way to do this without harding code the aggregation methods?

Reply:
pushed changes